### PR TITLE
[#577] Flow Web API: Handle Auth0 error response

### LINKED ISSUE #577: Flow Web API: Handle Auth0 error response

### PR DESCRIPTION
**Pull request by zuhdil**
Closes #577

### CODE REVIEW

**Review by MichaelAkvo — COMMENTED** [review]
One minimal suggestion for a different function to be used

File: sites/akvo-flow-web-api/app.py [hunk 1]
```diff
@@ -394,6 +396,25 @@ def check_password(rec):
     return False
 
 
+def timed_lru_cache(seconds, maxsize=128, typed=False):
+    def wrapper_cache(func):
+        f = lru_cache(maxsize, typed)(func)
+        f.lifetime = timedelta(seconds=seconds)
+        f.expiration = datetime.utcnow() + f.lifetime
```
[the remark this answers is not in the document]
Comment (zuhdil) [reply sites/akvo-flow-web-api/app.py]: Fixed

**Review by MichaelAkvo — APPROVED** [review]
LGTM

**Comment by MichaelAkvo** [comment]
Merged as it seems like more partners are affected by this. Additionally, I have pushed the new credentials. Hopefully that'll help

**Pull request merged**